test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -296,6 +296,46 @@ function ExecuteWebRequest
     return $result
 }
 
+# Starts the ClientCertificateCheck listener
+function StartClientCertificateCheck 
+{
+    param([int]$Port)
+    $initTimeoutSeconds  = 15
+    $projectRootPath     = Resolve-Path (Join-Path $PSScriptRoot "\..\..\..\..\")
+    $appSrcPath          = Join-Path $projectRootPath  'test\tools\ClientCertificateCheck'
+    $appPublishPath      = Join-Path $appSrcPath 'bin'
+    $appDllPath          = Join-Path $appPublishPath 'ClientCertificateCheck.dll'
+    $serverPfxPath       = Join-Path $appSrcPath 'ServerCert.pfx'
+    $serverPfxPassword   = 'password'
+    $initCompleteMessage = 'Now listening on'
+    
+    $timeOut = (get-date).AddSeconds($initTimeoutSeconds)
+    $Job = Start-Job {
```

Comment:
CI VMs is very-very slow. If locally you see no start delay then on CI it is possible the delay will be large. That's why I think it's best to use one tool (remove HttpListener) and run it only once.

>how to implement that

Call Invoke-WebRequest and analyze the response.